pkg/rke2/rke2_linux.go

```diff
@@ -199,3 +206,22 @@ func hostnameFQDN() (string, error) {
 
 	return strings.TrimSpace(b.String()), nil
 }
+
+func hostnameFromMetadataEndpoint() string {
+	resp, err := http.Get("http://169.254.169.254/latest/metadata/local-hostname")
```

Comment:
Access to the IMDS is usually done with a 1 second timeout to ensure that it doesn't hang indefinitely if the code isn't run on AWS. I think you need to pass through + wrap the context from initExecutor, and use `NewRequestWithContext` here instead of just Get.

Reply:
Updated.